services/persistence/main.py

```diff
@@ -73,96 +120,96 @@ def insert_into(database_filename, text_ids, labels, annotation_source):
-        FROM texts t
-        INNER JOIN labels l ON l.texts_id = t.id
-        ORDER BY RANDOM()
-        LIMIT ?""",
-        (n_texts, )
+            'statement': statement
+        } for text_id, statement, in c.execute(
+            """
+            SELECT
+                t.id text_id,
+                t.statement
+            FROM texts t
+            WHERE t.id IN({text_ids})""".format(text_ids=','.join('?'*len(ids))),
+            ids
         )
     ]
 
 
-def create_manifesto_storage(texts, labels):
+def get_texts_only(n_texts):
     """
-    creates required tables and inserts all of the given texts and labels.
+    return the text data only without labels, at most `n_texts`.
 
-    :param texts: iterable of political texts.
-    :param labels: iterable of political text labels.
+    :param n_texts: how many texts to retrieve, integer.
+    :return: majority voted labels per text: [
+        {'text_id': 1, 'statement': 'some statement', 'label': 'left'},
+        ...
+    ]
     """
     conn = sqlite3.connect(DB_FILENAME)
     c = conn.cursor()
+    results = [
+        {
+            'text_id': text_id,
+            'statement': statement
+        } for text_id, statement in c.execute(
+            """
+            SELECT
+                t.id text_id,
+                t.statement
+            FROM texts t
+            LIMIT ?""",
+            (n_texts, )
+        )
+    ]
+    return results
 
-    c.execute('DROP TABLE IF EXISTS texts')
-    c.execute(
-        '''CREATE TABLE texts (
-                id INTEGER PRIMARY KEY AUTOINCREMENT,
-                created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
-                statement TEXT NOT NULL
-            )
-        '''
-    )
-    c.execute('DROP TABLE IF EXISTS labels')
-    c.execute(
-        '''CREATE TABLE labels (
-                id INTEGER PRIMARY KEY AUTOINCREMENT,
-                texts_id INTEGER NOT NULL,
-                label INTEGER NOT NULL,
-                source text NOT NULL,
-                created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
-                FOREIGN KEY(texts_id) REFERENCES texts(id)
-            )
-        '''
-    )
-    conn.commit()
-    conn.close()
 
-    insert_into(DB_FILENAME, texts, labels, 'manifesto')
-
-    # for statement, label, source, text_created, label_created in c.execute(
-    #         """
-    #         SELECT
-    #             t.statement,
-    #             l.label,
-    #             l.source,
-    #             t.created_at,
-    #             l.created_at
-    #         FROM texts t
-    #         INNER JOIN labels l ON l.texts_id = t.id
-    #         LIMIT 10"""
-    # ):
-    #     print(statement, label, source)
-    #
-    # print(
-    #     'inserted',
-    #     c.execute("SELECT count(1) FROM texts").fetchone(),
-    #     'texts with',
-    #     c.execute("SELECT count(1) FROM labels").fetchone(),
-    #     'labels'
-    # )
-    #
-    # conn.close()
+def get_texts_with_majority_voted_labels(n_texts):
```

Comment:
man koennte auch alle, bei denen es ties gibt, weglassen - und so lange neu labeln lassen, bis es keine ties mehr gibt?

Reply:
Ja stimmt koennen wir auch so machen.
Hatte bloss ueberlegt, dass wenn wir das labelling starten, wird dann ein immer groesser werdender anteil der beispiele nur zwei labels haben die dann nicht mehr teil des trainings sind. was meinst du?

Reply:
stimmt. hm. vielleicht koennte man auch ohne label-aggregates arbeiten und bei ties einfach zwei verschiedene datenpunkte haben ... der classifer wuerde dann halt den label noise reflektieren. 

aber in jedem fall hast du recht und nen random sample bei ties nehmen macht am meisten sinn, alles zusammengenommen, also vor allem implementierungskomplexitaetsmaessig.

Reply:
dem modell einfach alle datenpunkte mitzugeben macht sinn finde ich. das ist sogar einfacher und damit dann nur noch ein `inner join` ohne majority vote.